### ISSUE TITLE
Pool and PoolAggregator objects require wrapping

### ISSUE THREAD

**Issue by nicb** [nicb/essentia-ruby#6]
However, they are complicate objects to wrap. They themselves need the `essentia::Mutex` object, which appears to be a mess.

**Comment by aterza** [comment]
Can the ruby own Mutex method be useful in some way?

**Comment by nicb** [comment]
No. You have no understanding of what `mutex` objects are. At least in this context.

**Comment by aterza** [comment]
I think I know what mutex objects are, we've also discussed them. If the ruby own Mutex method cannot be useful in any way, in this context, I think it's another matter.
